### PR TITLE
TOC on Friday, wat!

### PR DESCRIPTION
**Pull request by jaketrent**
- A little stinky to have to write it in every mdx file, but that's to allow for inline placement per the mobile design
- But inline is nice, because it allows custom placement
- Also learned that full component imports can be done with shortcodes

### CODE REVIEW

**Review by danethurber — COMMENTED** [review]
i'm getting this error. i've done a gatsby clean and am still getting the error

```
 ERROR #98124  WEBPACK

Generating SSR bundle failed

Can't resolve '../../../../src/components/component' in '/Users/dane/Projects/design-system/docs/.cache/caches/gatsby-plugin-mdx/mdx-scopes-dir'

If you're trying to use a package make sure that '../../../../src/components/component' is installed. If you're trying to use a local file make sure that the path is correct.
```

File: docs/src/components/mdx/index.tsx [hunk 1]
```diff
@@ -3,13 +3,18 @@ import DSLink from '@pluralsight/ps-design-system-link'
 import * as Text from '@pluralsight/ps-design-system-text'
 import cx from 'classnames'
 import { Link as GatsbyLink } from 'gatsby'
-import React, { HTMLAttributes } from 'react'
+import React, { AllHTMLAttributes, HTMLAttributes } from 'react'
```
Comment (danethurber) [inline docs/src/components/mdx/index.tsx:6]: what's the diff between AllHTMLAttributes and HTMLAttributes

File: docs/src/components/table-of-contents.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+interface TableOfContentsProps {
+  headings: Heading[]
+}
+export const TableOfContents: React.FC<TableOfContentsProps> = props => {
+  const themeName = useTheme()
+  let slugger = new GithubSlugger()
+  React.useEffect(() => {
+    slugger = new GithubSlugger()
+  }, [slugger])
```
Comment (danethurber) [inline docs/src/components/table-of-contents.tsx:26]: you could use a ref for this but i'm not sure it matters

File: docs/src/components/table-of-contents.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  return (
+    <nav
+      className={styles.toc}
+      style={
+        {
+          '--bg':
+            themeName === Theme.names.dark
+              ? colorsBackgroundUtility[25]
+              : colorsBackgroundLight[2]
+        } as React.CSSProperties
```
Comment (danethurber) [inline docs/src/components/table-of-contents.tsx:36]: is this a new pattern we should follow. using css vars instead of toggling classes

Comment (jaketrent) [reply docs/src/components/table-of-contents.tsx:36]: Oh, I'm not sure.  I was just geeking out about it.  Had seen it in the wild but never in a project like ours until types/prop.tsx was updated.

I think I still prefer class toggling.  Keeps the core out of the code and the style info in a stylesheet.

Do you think we should consolidate approaches?

File: docs/src/components/table-of-contents.tsx (hunk 2 above)
Comment (jaketrent) [reply docs/src/components/table-of-contents.tsx:26]: Seems to work as-is.  I avoid refs if I can.

File: docs/src/components/mdx/index.tsx (hunk 1 above)
Comment (jaketrent) [reply docs/src/components/mdx/index.tsx:6]: One screams about not having `href`, and the other one's ok.

File: docs/src/components/table-of-contents.tsx (hunk 3 above)
Comment (danethurber) [reply docs/src/components/table-of-contents.tsx:36]: personally i'm not to worried about consolidating. let's see how it goes

**Review by danethurber — CHANGES_REQUESTED** [review]
i'm having issues on the component pages - they aren't loading

also i just want to double check this is the TOC experience on mobile that we want...

Screen Shot 2020-09-28 at 11 34 22 AM

or do we have a mobile optimized design

**Comment by jaketrent** [comment]
@danethurber I double-checked the mobile design and made one adjustment.

Regarding the pages not working, it seems as thought Types is not available through the MDXProvider#components shortcodes.  The amazing thing is that this has been working twice before but I've somehow bungled it twice since then. :) Anyway, I'll followup on the Types provider problem.

**Comment by jaketrent** [comment]
Hey @danethurber , I think the 2 issues are addressed:

- Mobile design dialed
- Types is now available as an mdx shortcut. Had to change naming convention away from the dot naming.  Eg, Types.Table becomes TypesTable.

Please re-review.

**Review by danethurber — APPROVED** [review]
do it live

**Pull request merged**